Source/source/GPS.cpp

```diff
@@ -9,21 +9,23 @@ GPS_Data_Types data_types;
 //checks if GPS is turned on, if not, turn on. 
 //returns TRUE if successful, FALSE otherwise
 bool init_gps() {
-	if (!gpsOn) {
-		//if can turn on gps
-		gpsOn = true;
-		return true;
-		//if cant turn on gps, return false
-	}
-	return true;
+    if (!gpsOn) {
+        //if can turn on gps
+        gpsOn = true;
+        return true;
+        //if cant turn on gps, return false
+    }
+    return true;
 }
 
+
 //polls the GPS, gets data if turned on
-string poll() {
-	if (gpsOn) {
-		return get_message();
-	}
-	return NULL;
+bool poll(string *message) {
+    if (init_gps()) {
+        cout << "GPS on" << endl;
+        return read_nmea_from_file(message);
+    }
+    return false;
```

Comment:
I agree with this. Hopefully the hardware will give us an indicator that it failed the polling. Otherwise, fault checking code will have to be implemented here (something Kaylee is working on) and use the result of that as the return.